internal/backend/remote/backend.go

```diff
@@ -716,6 +709,28 @@ func (b *Remote) StateMgr(ctx context.Context, name string) (statemgr.Full, erro
 	}, nil
 }
 
+func localVersionMatchesRemoteVersion(wsv string) error {
+	const latestVersion = "latest"
+
+	if wsv == latestVersion {
+		// We handle the "latest" for compatibility reasons:
+		// TFE and potentially other backends can set the workspace version to "latest"
+		// if they were created using pre-released version of terraform
+		return nil
+	}
+
+	// We trim the suffix if the backend workspace contains it.
+	wsv = version.Must(version.NewVersion(wsv)).Core().String()
```

Comment:
In theory, we're trusting that the version returned here is either "latest" or a valid semver

By calling `Must` here will cause a panic if for some reason the remote backend has returned a non-valid, non-latest version.

In my opinion, I'd like to fail a little bit more gracefully and return the error and ensure that it's bubbled and handled higher up.